app/src/main/java/ca/pkay/rcloneexplorer/Database/xml/Exporter.java

```diff
@@ -0,0 +1,129 @@
+package ca.pkay.rcloneexplorer.Database.xml;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.os.Environment;
+import android.support.v4.app.ActivityCompat;
+import android.widget.Toast;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.StringWriter;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import ca.pkay.rcloneexplorer.Database.DatabaseHandler;
+import ca.pkay.rcloneexplorer.Database.Task;
+import ca.pkay.rcloneexplorer.R;
+import es.dmoral.toasty.Toasty;
+
+public class Exporter {
+
+    private static final int PERMISSION_WRITE_EXTERNAL = 739;
+
+    public static void export(Activity activity){
+        try {
+            String xml = Exporter.create(activity);
+            Exporter.storeFile(activity, xml);
+        } catch (ParserConfigurationException | TransformerException | IOException e) {
+            e.printStackTrace();
```

Comment:
Always use [FLog](https://github.com/x0b/rcx/blob/master/app/src/main/java/ca/pkay/rcloneexplorer/util/FLog.java) with appropriate log level and log tag (create log tags with ``logt`` shortcut). E.g. ``FLog.e(TAG, "Error when exporting tasks", e)``